packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.jsx

```diff
@@ -56,28 +57,36 @@ export const withReactQuery = (Component) => {
      * @return {Promise<Object[]>}
      */
     WrappedComponent.getDataPromises = (renderContext) => {
-        const {AppJSX} = renderContext
+        const {AppJSX: OriginalApp} = renderContext
+
+        renderContext.AppJSX = React.createElement(
+            IsPrePassContext.Provider,
+            {value: true},
+            OriginalApp
+        )
 
         const dataPromise = Promise.resolve()
-            .then(() => ssrPrepass(AppJSX)) // NOTE: ssrPrepass will be included in the vendor bundle. BAD
+            .then(() => ssrPrepass(renderContext.AppJSX)) // NOTE: ssrPrepass will be included in the vendor bundle. BAD
```

Comment:
When running pre-pass, `AppJSX` would be wrapped with pre-pass context set to true. This context is how `useServerContext` can know whether it's being run during pre-pass or not.

Reply:
On line 62, I couldn't use plain JSX because I believe it wasn't parsed, since it was not inside a component function (it's inside `WrappedComponent.getDataPromises`). So I had to use the lower-level `React.createElement` instead.

Reply:
would something like this work?

```
// outside WrappedComponent.getDataPromises

function withPrePassContext(Wrapped) {
   return function (props) {
      return <IsPrePassContext.Provider><Wrapped {...props}/></IsPrePassContext.Provider>
   }
}

// in WrappedComponent.getDataPromises
const dataPromise = Promise.resolve()
            .then(() => ssrPrepass(withPrePassContext(renderContext.AppJSX)))

```

Reply:
@kevinxh unfortunately, it still does not work. With `React.isValidElement`, I've compared both the original App and the resulting wrapped component, and the latter is not considered to be a valid React element.